### ISSUE TITLE
[Bug Report] MessagingClient.sendAsync fails silently after a client reconnect event

### ISSUE THREAD

**Issue by luke-sun-tw** [Azure/azure-iot-sdk-java#1728]
We recently experienced a live outage and inspecting the log confirms that the client experienced a network outage and was closed. Our app then calls MessagingClient.open() to reconnect the client, which succeeded. 

Moments later our app resends failed messages using MessagingClient.sendAsync method:

```        
try
        {
            client.sendAsync(deviceId, message, messageSentCallback, null);
        }
        catch (IllegalStateException e)
        {
            log.info("Client was closed when attempting to send a message. Re-opening the client and trying again");
            open(); // <--- calls MessagingClient.open() in a retry loop
            if(connectionStatus == ConnectionStatus.CONNECTED)
            {
                sendAsync(deviceId, message); // <-- recursively calls this method
            }
            else
            {
                log.warn("Message {} could not be sent because MessagingClient was not open", message.getMessageId());
            }
        }
```

But there was no evidence of messageSentCallback ever being invoked or IllegalStateException exception being thrown. MessagingClient remained in this state for several days.

Environment:

iot-service-client version: 2.1.5
Base container image: azul/zulu-openjdk-alpine:17

**Comment by timtay-microsoft** [comment]
Do you have any SDK logs from around this time that you could share?

**Comment by PashaKonietin** [comment]
Hello, we have the very similar situation.
Previously we had a pool of opened ServiceClient's (iot-service-client of version 1.34.2) that were used to send cloud to device messages.
With the update of iot-service-client library to version 2.1.6 we started to receive lots of exceptions like "Client must be opened before any message can be sent", almost right after the start of the application, or after ~30 minutes, even though all the Messaging clients were initialised and opened successfully. With older version of library none of exception of this kind were noticed.  Ti fix this problem we added the "reopen" in the retry section(similar to what is done in @luke-sun-tw code), the only difference is that we are using synchronous send() method. But the symptoms are the same, messages are not getting to the device, and MessagingClient connection continuously closes up.
Unfortunately, we do not have any logs, as this is a hight loaded application, so there is no possibility to get the trace logs from the sdk. Not reproduced locally either, yet.
